File: src/array/array_java/Ch9_CeilAndFloor.java
package array.array_java;

import java.util.Scanner;

public class Ch9_CeilAndFloor {

    public static void main(String[] args) {
        Scanner scn = new Scanner(System.in);

        int n = scn.nextInt();
        int[] arr = new int[n];

        for (int i = 0; i < arr.length; i++) {
            arr[i] = scn.nextInt();
        }

        int data = scn.nextInt();
        int lo = 0;
        int hi = arr.length - 1;
        int ceil = 0;
        int floor = 0;
        while (lo <= hi) {
            int mid = (lo + hi) / 2;
            if (data > arr[mid]) {
                lo = mid + 1;
                floor = arr[mid];
            } else if (data < arr[mid]) {
                hi = mid - 1;
                ceil = arr[mid];
            } else {
                ceil = arr[mid];
                floor = arr[mid];
            }
        }

        System.out.println("Ceil is " + ceil + " Floor is " + floor);
    }
}

File: src/array/array_kotlin/Ch9_CeilAndFloor.kt
package array.array_kotlin

import java.util.*

fun main() {
    ceilAndFloor()
}

fun ceilAndFloor() {
    val scn = Scanner(System.`in`)

    val n = scn.nextInt()
    val arr = IntArray(n)

    for (i in arr.indices) {
        arr[i] = scn.nextInt()
    }

    val data = scn.nextInt()
    var lo = 0
    var hi = arr.size - 1
    var ceil = 0
    var floor = 0
    while (lo <= hi) {
        val mid = (lo + hi) / 2
        if (data > arr[mid]) {
            lo = mid + 1
            floor = arr[mid]
        } else if (data < arr[mid]) {
            hi = mid - 1
            ceil = arr[mid]
        } else {
            ceil = arr[mid]
            floor = arr[mid]
        }
    }
    println("Ceil is $ceil Floor is $floor")
}
